src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -2481,7 +2869,25 @@ public function class_list() {
 	 * @return string
 	 */
 	public function get_modifiable_text() {
-		return $this->is_virtual() ? '' : parent::get_modifiable_text();
+		if ( $this->is_virtual() ) {
+			return '';
+		}
+
+		$text = parent::get_modifiable_text();
+		if ( ! str_contains( $text, "\x00" ) ) {
+			return $text;
+		}
+
+		if ( '#comment' === $this->get_token_name() ) {
```

Comment:
Funky comments probably need this handling too?

Reply:
maybe? I want to audit this because the rules are all different for various contexts and insertion modes, mainly whether null bytes transform into U+FFFD or whether it's stripped away. I'll follow-up with this, but I don't feel like it has to be in this patch

Reply:
Updated to handle all situations, which I think is right now.

I also moved the logic into the Tag Processor, because it seems like it's capable enough to know whether to skip the newline. Turns out that if it's _anything_ other than a text node immediately following a PRE or LISTING (since TEXTAREA is self-contained, the logic there is clear), then there's no skipping. A simple flag `WP_HTML_Tag_Processor::skip_next_line_feed` suffices.